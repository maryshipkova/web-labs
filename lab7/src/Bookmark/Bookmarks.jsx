import React from 'react';
import './Bookmarks.scss'
import {connect} from "react-redux";
import {add} from '../store/actions.js';

const BookmarksFC = (props) => {
    const [value, setValue] = React.useState('');
    const handleSubmit = (e) => {
        e.preventDefault();
        props.add(value);
    };

    // избранное в localStorage
    return (
        <main className={'Bookmarks'}>
            <div className={'Bookmarks-Row'}>
                <h3 className={'Bookmarks-Title'}>Избранное</h3>
                <form onSubmit={handleSubmit} className={'Bookmarks-Add'}>
                    <input
                        className={'Bookmarks-Input'}
                        type="text"
                        placeholder={'Добавить новый город'}
                        onChange={e => setValue(e.target.value)}
                    />
                    <input type={"submit"} className={'Bookmarks-AddButton'} value={'+'}/>
                </form>
            </div>
        </main>
    )
};


const mapDispatchToProps = {
    add
};

export const Bookmarks = connect(null, mapDispatchToProps)(BookmarksFC);
