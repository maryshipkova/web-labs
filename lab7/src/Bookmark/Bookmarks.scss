@import '../constants.scss';

.Bookmarks {
  &-Row{
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &-Input{

  }

  &-AddButton{
    margin-left: 10px;
  }

}
