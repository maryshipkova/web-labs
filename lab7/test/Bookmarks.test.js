import React from 'react';
import {Bookmarks} from "../src/Bookmarks/Bookmarks";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";

it('Bookmarks', () => {
    const initialState={cities: ["Moscow","1234","12345"]};
    const mockStore = configureStore([thunk]);
    const store = mockStore(initialState);

    const tree = renderer
        .create(
            <Provider store={store}>
                <Bookmarks/>
            </Provider>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
