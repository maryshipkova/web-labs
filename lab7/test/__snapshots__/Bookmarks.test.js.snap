// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bookmarks 1`] = `
<main
  className="Bookmarks"
>
  <div
    className="Bookmarks-Row"
  >
    <h3
      className="Bookmarks-Title"
    >
      Избранное
    </h3>
    <form
      className="Bookmarks-Add"
      onSubmit={[Function]}
    >
      <input
        className="Bookmarks-Input"
        onChange={[Function]}
        placeholder="Добавить новый город"
        type="text"
      />
      <input
        className="Bookmarks-AddButton Weather-Button"
        type="submit"
        value="+"
      />
    </form>
  </div>
</main>
`;
